### DISCUSSION TITLE
Customer portal

### DISCUSSION THREAD

**Discussion by jozefrebjak** [codeigniter4/shield#243] [Q&A]
Hello, I need to setup customer portal, we already have a table customers with customer details. 

Can we discuss how to get working customer portal with Shield. 

I can imagine that we need to create user for all of the customers (or not?) to allow them to access app. 

My plan is to add them to the group called for example `customers` and then redirect them to the different namespace for example `/customers`. 

Is this a way how to achieve such a goal ?

Question is do we need to create usernames and passwords for all of them or just use some magic link. 

There can be thousands of the customers. 

Thank you.

**Comment by kenjis** [comment]
> Question is do we need to create usernames and passwords for all of them or just use some magic link.

You need to create emails (and usernames) and passwords for all of them.

The magic link is not for normal login. 
It is for when a user forgets his/her password.
If an attacker can see the email with magic link, they can login as the user.
